src/QMCWaveFunctions/lcao/SoaCuspCorrectionBasisSet.h

```diff
@@ -145,10 +166,7 @@ struct SoaCuspCorrection
    */
   inline void evaluateVGL(const ParticleSet& P, int iat, VGLVector_t& vgl)
   {
-
-    constexpr RealType czero(0);
-
-    myVGL=czero;
+    myVGL=0.0;
```

Comment:
More of a personal taste. I prefer RealType(0) instead of 0.0. Complier should turn RealType(0)  to 0.0 or 0.0f based on RealType.

Reply:
The compiler produces exactly the same instructions either way. I took this to be an attempt to write generic code that would support complex types.
Only in that case is T t = T(0) different than writing (T t = 0 | T t = 0.0 | T t = 0.0f); 

The only time you need to write 1.0f instead of 1 for a literal constant is if you have something like
T t = 1 / 2; In modern C++, definitely not true in Fortran.

Reply:
IMO either are fine. I tend to prefer 0.0 since it is clear that a floating point number is intended when the source is read. If we truly think it matters, the coding guide should be updated first.